t/02_yaml_schedule.t

```diff
@@ -87,4 +87,44 @@ subtest 'parse_yaml_test_schedule_recursive_conditional' => sub {
     ok $modules[3] eq 'bar/test3', "Basic scheduling";
 };
 
+subtest 'merge default schedules with flows and individual schedules' => sub {
+    use scheduler;
+    use testapi 'set_var';
+
+    # positive unit tests
+
+    my $schedule = $ypp->load_file(dirname(__FILE__) . '/data/test_data_mini_default_flows.yaml');
+    my @modules;
+    set_var('YAML_SCHEDULE_DEFAULT', '/t/data/flows/default.yaml');
+    set_var('YAML_SCHEDULE_FLOWS', 'flow_a');
+    @modules = scheduler::parse_schedule($schedule);
+    ok $modules[0] eq 'flow_a_foo', "Merge default with flows and individual";
+    ok $modules[1] eq 'flow_a_bar', "Merge default with flows and individual";
+    ok $modules[2] eq 'individual_step_3', "Merge default with flows and individual";
+    set_var('YAML_SCHEDULE_FLOWS', 'flow_b');
+    @modules = scheduler::parse_schedule($schedule);
+    ok $modules[0] eq 'flow_b_foo', "Merge default with flows and individual";
+    ok $modules[1] eq 'flow_b_bar', "Merge default with flows and individual";
+    ok $modules[2] eq 'individual_step_3', "Merge default with flows and individual";
+    set_var('YAML_SCHEDULE_FLOWS', 'flow_a,flow_b');    # flow_b should win
+    @modules = scheduler::parse_schedule($schedule);
+    ok $modules[0] eq 'flow_b_foo', "Merge default with flows and individual";
+    ok $modules[1] eq 'flow_b_bar', "Merge default with flows and individual";
+    ok $modules[2] eq 'individual_step_3', "Merge default with flows and individual";
+    set_var('YAML_SCHEDULE_FLOWS', 'flow_b,flow_a');    # flow_a should win
+    @modules = scheduler::parse_schedule($schedule);
+    ok $modules[0] eq 'flow_a_foo', "Merge default with flows and individual";
+    ok $modules[1] eq 'flow_a_bar', "Merge default with flows and individual";
+    ok $modules[2] eq 'individual_step_3', "Merge default with flows and individual";
+
+    # negative unit tests
+    set_var('YAML_SCHEDULE_DEFAULT', '/t/data/flows/default_broken.yaml');
+    dies_ok { scheduler::parse_schedule($schedule) } 'Expected to see an error here';
+
+    set_var('YAML_SCHEDULE_DEFAULT', '/t/data/flows/default.yaml');
+    set_var('YAML_SCHEDULE_FLOWS', 'flow_a,nonexistent_flow');
```

Comment:
same here, this is not a negative case, it shouldn't happen, we should check existence of files, and in fact we do, we added that to our ci.
let me think about negative cases, if there is any...